repose-aggregator/components/filters/ip-classification/src/main/scala/org/openrepose/filters/ipclassification/IPClassificationFilter.scala

```diff
@@ -0,0 +1,153 @@
+  private val cidrList = new ConcurrentLinkedQueue[CIDRTuple]()
+  private var groupHeaderName: String = _
+  private var groupHeaderQuality: Double = _
+  private var userHeaderName: String = _
+  private var userHeaderQuality: Double = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    config = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+    logger.info(s"Initializing using config $config")
+    val xsdURL: URL = getClass.getResource("/META-INF/config/schema/ip-classification.xsd")
+
+    configurationService.subscribeTo(
+      filterConfig.getFilterName,
+      config,
+      xsdURL,
+      this,
+      classOf[IpClassificationConfig]
+    )
+  }
+
+  override def destroy(): Unit = configurationService.unsubscribeFrom(config, this.asInstanceOf[UpdateListener[_]])
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    new FilterLogicHandlerDelegate(servletRequest, servletResponse, filterChain).doFilter(this)
+  }
+
+  override def handleRequest(httpServletRequest: HttpServletRequest, httpServletResponse: ReadableHttpServletResponse): FilterDirector = {
+    logger.debug("Handling HTTP request!")
+    val filterDirector = new FilterDirectorImpl()
+
+    getClassificationLabel(httpServletRequest.getRemoteAddr).foreach { label =>
+      filterDirector.requestHeaderManager.appendHeader(groupHeaderName, label, groupHeaderQuality)
+    }
+
+    //Always set the user header name to the current IP address
+    //TODO: apparently we append users as well?
+    filterDirector.requestHeaderManager().appendHeader(userHeaderName, httpServletRequest.getRemoteAddr, userHeaderQuality)
+
+    filterDirector.setFilterAction(FilterAction.PASS) //Don't need to process a response with this filter, so ship it
+    filterDirector
+  }
+
+  def getClassificationLabel(ipAddress: String): Option[String] = {
+    import scala.collection.JavaConversions._
+    cidrList.toList.find { case (_, cidrUtil) =>
+      cidrUtil.isInRange(ipAddress)
+    } map { case (label, _) =>
+      label
+    }
+  }
+
+  override def configurationUpdated(config: IpClassificationConfig): Unit = {
+    //Update all the CIDRs
+    //Create a new list to replace the old one
+    val items = new util.ArrayList[CIDRTuple]()
+
+    import scala.collection.JavaConversions._
+    val classifications = config.getClassifications.getClassification.toList
+    classifications.foreach { classification =>
+      val label = classification.getLabel
+      def splitCIDR(javaCIDR: String): List[String] = {
+
+        Option(javaCIDR).map { cidr =>
+          cidr.split(" ").toList
+        } getOrElse {
+          List.empty[String]
+        }
+      }
+
+      splitCIDR(classification.getIpv4Cidr).map { cidr =>
+        items.add((label, new CIDRUtils(cidr)))
+      }
+      splitCIDR(classification.getIpv6Cidr).map { cidr =>
+        items.add((label, new CIDRUtils(cidr)))
+      }
```

Comment:
Because we specify both ipv4 and ipv6. As well there's a TODO listed about how an IPv6 catchall will match an IPv4 address. So, if that needs to be changed, I left the logic paths separate. Otherwise, we could probably do some fancyness to smush them together, but they're originally treated as two separate entities.